src/packages/recompose/renderNothing.js

```diff
@@ -1,9 +1,14 @@
+import { Component } from 'react'
 import createHelper from './createHelper'
 
-const renderNothing = _ => {
-  const Nothing = () => null
-  Nothing.displayName = 'Nothing'
-  return Nothing
+class Nothing extends Component {
+  render() {
+    return null
+  }
 }
```

Comment:
As I know big developer daddies always try to avoid third party libraries version check like `if (version.startsWith('0.14')) {` ;-)

Why do you think that unused part will be removed at production? 
My usual build tools `babel + webpack + UglifyJsPlugin` does not remove anything in such case